lib/BC/SleighLifter.cpp

```diff
@@ -404,70 +419,30 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
-              bldr.GetInsertBlock()->getModule(), llvm::Intrinsic::sqrt);
-          llvm::Value *sqrt_args[] = {*sqrt_inval};
-          return this->LiftStoreIntoOutParam(
-              bldr, bldr.CreateCall(sqrt_intrinsic, sqrt_args), outvar);
-        }
-        break;
+        return this->LiftUnOpWithFloatIntrinsic(bldr, llvm::Intrinsic::sqrt,
+                                                outvar, input_var);
       }
       case OpCode::CPUI_FLOAT_CEIL: {
-        auto ceil_inval = this->LiftInParam(
-            bldr, input_var, llvm::Type::getFloatTy(this->context));
-        if (ceil_inval.has_value()) {
-          llvm::Function *ceil_intrinsic = llvm::Intrinsic::getDeclaration(
-              bldr.GetInsertBlock()->getModule(), llvm::Intrinsic::ceil);
-          llvm::Value *ceil_args[] = {*ceil_inval};
-          return this->LiftStoreIntoOutParam(
-              bldr, bldr.CreateCall(ceil_intrinsic, ceil_args), outvar);
-        }
-        break;
+        return this->LiftUnOpWithFloatIntrinsic(bldr, llvm::Intrinsic::ceil,
+                                                outvar, input_var);
       }
       case OpCode::CPUI_FLOAT_FLOOR: {
-        auto floor_inval = this->LiftInParam(
-            bldr, input_var, llvm::Type::getFloatTy(this->context));
-        if (floor_inval.has_value()) {
-          llvm::Function *floor_intrinsic = llvm::Intrinsic::getDeclaration(
-              bldr.GetInsertBlock()->getModule(), llvm::Intrinsic::floor);
-          llvm::Value *floor_args[] = {*floor_inval};
-          return this->LiftStoreIntoOutParam(
-              bldr, bldr.CreateCall(floor_intrinsic, floor_args), outvar);
-        }
-        break;
+        return this->LiftUnOpWithFloatIntrinsic(bldr, llvm::Intrinsic::floor,
+                                                outvar, input_var);
       }
       case OpCode::CPUI_FLOAT_ROUND: {
-        auto round_inval = this->LiftInParam(
-            bldr, input_var, llvm::Type::getFloatTy(this->context));
-        if (round_inval.has_value()) {
-          llvm::Function *round_intrinsic = llvm::Intrinsic::getDeclaration(
-              bldr.GetInsertBlock()->getModule(), llvm::Intrinsic::round);
-          llvm::Value *round_args[] = {*round_inval};
-          return this->LiftStoreIntoOutParam(
-              bldr, bldr.CreateCall(round_intrinsic, round_args), outvar);
-        }
-        break;
+        return this->LiftUnOpWithFloatIntrinsic(bldr, llvm::Intrinsic::round,
+                                                outvar, input_var);
       }
       case OpCode::CPUI_FLOAT_NAN: {
         auto nan_inval = this->LiftInParam(
             bldr, input_var, llvm::Type::getFloatTy(this->context));
         if (nan_inval.has_value()) {
-          // LLVM trunk has an `isnan`intrinsic but to support older versions, I think we need to do this.
+          // LLVM trunk has an `isnan` intrinsic but to support older versions, I think we need to do this.
```

Comment:
Is it supported in LLVM 13 or 14? If 14 then we'll move to that reasonably soon, so file an issue to add its use in.

Reply:
Oh, interesting. The intrinsic got reverted in https://github.com/llvm/llvm-project/commit/3f1f08f0ed6aa1ebd4678583cbbf5026ca5cbdf8.

I'll still make an issue to check whether it makes its way into 15.